encoder.py

```diff
@@ -0,0 +1,106 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+import os
+from typing import Optional, List
+
+import numpy as np
+
+from jina import Executor, requests, DocumentArray
+
+import torch
+import torchvision.models as models
+from torch.hub import load_state_dict_from_url
+
+
+class ImageTorchEncoder(Executor):
+    """
+    :class:`ImageTorchEncoder` encodes ``Document`` content from a ndarray,
+    potentially B x (Channel x Height x Width) into a ndarray of `B x D`.
+    Where B` is the batch size and `D` is the Dimension.
+    Internally, :class:`ImageTorchEncoder` wraps the models from `
+    `torchvision.models`.
+    https://pytorch.org/docs/stable/torchvision/models.html
+    :param model_name: the name of the model. Supported models include
+        ``resnet18``, ``alexnet``, `squeezenet1_0``,  ``vgg16``,
+        ``densenet161``, ``inception_v3``, ``googlenet``,
+        ``shufflenet_v2_x1_0``, ``mobilenet_v2``, ``resnext50_32x4d``,
+        ``wide_resnet50_2``, ``mnasnet1_0``
+    :param pool_strategy: the pooling strategy. Options are:
+        - `None`: Means that the output of the model will be the 4D tensor
+            output of the last convolutional block.
+        - `mean`: Means that global average pooling will be applied to the
+            output of the last convolutional block, and thus the output of
+            the model will be a 2D tensor.
+        - `max`: Means that global max pooling will be applied.
+    :param channel_axis: The axis of the color channel, default is 1
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+    DEFAULT_TRAVERSAL_PATH = ['r']
+
+    def __init__(
+        self,
+        model_name: str = 'mobilenet_v2',
+        pool_strategy: str = 'mean',
+        channel_axis: int = 1,
+        load_pre_trained_from_path: Optional[str] = None,
+        default_traversal_path: Optional[List[str]] = None,
```

Comment:
add docstrings for these two

Reply:
https://github.com/jina-ai/executor-image-torch-encoder/pull/1/commits/a5aeb03fded5c8d2c3bf8d2220a27eef5a73619d